### DISCUSSION TITLE
go-ping: Error listening for ICMP packets: socket: permission denied

### DISCUSSION THREAD

**Discussion by Abdalnablse10** [netblue30/firejail#5774] [Q&A] [answered]
A program I'm currently running uses go-ping, it normally works without firejail "no root by the way" but gives this error when using firejail : Error listening for ICMP packets: socket: permission denied
What should i do to allow it?

Edit: I'm also using --net=eth0

**Comment by ghost** [comment]
How are you running this go-ping app? With `--noprofile` (implies default.profile)? Or did you create a custom profile for it (show us if you did)?

You could try using the ping.profile as base. IMO the main option to allow listening to ICMP packets is `caps.keep net_raw`. Depending on whether go-ping needs to raise privileges, make sure to avoid `nonewprivs`, `noroot` and possibly `protocol`.

**Reply by Abdalnablse10** [reply]
At first i use "firejail --net=eth0 saidprogram"
But got the permission error, so i tried to use --no-profile but made no difference.

**Comment by kmk3** [comment, marked as answer]
Closing in favor of:

* #5775

**Comment by kmk3** [comment]
(Offtopic)

@Abdalnablse10

If you want to report another bug/issue with a program, please open an issue
and follow the bug report template:

* <https://github.com/netblue30/firejail/issues/new?template=bug_report.md>

Bugs are tracked in the issue tracker; discussions have poor UX for this.  The
latter are intended for things like general discussion and Q&A.
